stream-chat-android-ui-common/src/main/kotlin/io/getstream/chat/android/ui/common/state/messages/list/ChannelHeaderViewState.kt

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2014-2025 Stream.io Inc. All rights reserved.
+ *
+ * Licensed under the Stream License;
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://github.com/GetStream/stream-chat-android/blob/main/LICENSE
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.getstream.chat.android.ui.common.state.messages.list
+
+import io.getstream.chat.android.core.ExperimentalStreamChatApi
+import io.getstream.chat.android.models.Channel
+import io.getstream.chat.android.models.ConnectionState
+import io.getstream.chat.android.models.User
+
+/**
+ * Represents the state of the channel header in the UI.
+ */
+@ExperimentalStreamChatApi
+public data class ChannelHeaderViewState(
+    /**
+     * The current connected user.
+     */
+    val currentUser: User? = null,
+
+    /**
+     * The connection state of the chat client.
+     */
+    val connectionState: ConnectionState = ConnectionState.Offline,
+
+    /**
+     * The channel associated with the message list.
+     */
+    val channel: Channel = Channel(),
```

Comment:
Do we need this default value?
Does it make sense to create `ChannelHeaderViewState` with this default value? :thinking:

Reply:
That was to prevent nullables, as the legacy components expect non-nulls.

<img width="328" alt="Screenshot 2025-05-22 at 08 28 28" src="https://github.com/user-attachments/assets/10c7529f-7f86-45d0-990e-c2a4435940df" />

The goal of the channel header view state/controller was not to refactor the legacy components, but to decouple the state from the other view controllers.

Reply:
What I mean is, Do we need to init it with a default value?
We can keep `channel` as non-nullable property, but asking to be initialized outside, not by this default argument